src/sentry/models/releasefile.py

```diff
@@ -38,15 +40,20 @@ class Meta:
 
     def save(self, *args, **kwargs):
         if not self.ident and self.name:
-            self.ident = type(self).get_ident(self.name)
+            dist = self.dist and self.dist.name or None
+            self.ident = type(self).get_ident(self.name, dist)
         return super(ReleaseFile, self).save(*args, **kwargs)
 
     def update(self, *args, **kwargs):
         # If our name is changing, we must also change the ident
         if 'name' in kwargs and 'ident' not in kwargs:
-            kwargs['ident'] = self.ident = type(self).get_ident(kwargs['name'])
+            dist = kwargs.get('dist') or self.dist
+            kwargs['ident'] = self.ident = type(self).get_ident(
+                kwargs['name'], dist and dist.name or dist)
         return super(ReleaseFile, self).update(*args, **kwargs)
 
     @classmethod
-    def get_ident(cls, name):
+    def get_ident(cls, name, dist=None):
+        if dist is not None:
+            return sha1_text(name + '@@' + dist).hexdigest()
```

Comment:
Hopefully this never causes a problem in reality. It might be hard to debug. Since in theory, this conflicts with any release that has '@@' in it. Maybe we can use a character that's invalid in a release name instead to guarantee they don't overlap.